app/_how-tos/use-ai-azure-content-safety-output-mode.md

```diff
@@ -0,0 +1,264 @@
+
+{% entity_examples %}
+entities:
+    plugins:
+    - name: ai-proxy
+      config:
+        route_type: llm/v1/chat
+        auth:
+          header_name: Authorization
+          header_value: Bearer ${openai_api_key}
+        model:
+          provider: openai
+          name: gpt-4o
+variables:
+  openai_api_key:
+    value: $OPENAI_API_KEY
+{% endentity_examples %}
+
+## Configure the AI Azure Content Safety plugin
+
+In this example, the plugin protects against harmful *model outputs* rather than [user inputs](/how-to/use-azure-ai-content-safety/).
+The configuration uses Azure’s four harm categories and instructs Kong to evaluate every response before returning it to the client.
+
+In this configuration we:
+* Set `guarding_mode: OUTPUT` to inspect and filter **responses from the LLM**.
+* Map Azure harm categories (`Hate`, `SelfHarm`, `Sexual`, and `Violence`) under `categories[]`.
+* Use `rejection_level: 2` to block moderately harmful content or higher.
+* Set `output_type: FourSeverityLevels` for a 1–4 severity scale.
+* Enable `reveal_failure_reason: true` to include a detailed reason when content is rejected.
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: ai-azure-content-safety
+      config:
+        content_safety_url: ${azure_content_safety_url}
+        content_safety_key: ${azure_content_safety_key}
+        categories:
+          - name: Hate
+            rejection_level: 2
+          - name: SelfHarm
+            rejection_level: 2
+          - name: Sexual
+            rejection_level: 2
+          - name: Violence
+            rejection_level: 2
+        text_source: concatenate_user_content
+        reveal_failure_reason: true
+        output_type: FourSeverityLevels
+        guarding_mode: OUTPUT
+variables:
+  azure_content_safety_key:
+    value: $AZURE_CONTENT_SAFETY_KEY
+  azure_content_safety_url:
+    value: $AZURE_CONTENT_SAFETY_URL
+{% endentity_examples %}
+
+{:.warning}
+> Ensure that `$DECK_AZURE_CONTENT_SAFETY_URL` points to the `/contentsafety/text:analyze` endpoint.
+
+## Test the configuration
+
+Send a chat request that triggers a harmful model response.
+In this test, the user asks the model to generate violent or hateful content:
+
+{% navtabs "azure-content-safety-tests" %}
+
+{% navtab "Denied: Hate" %}
+
+This test provokes hateful output from the model and should be blocked by the OUTPUT guard:
+
+<!--vale off-->
+{% validation request-check %}
+url: /anything
+headers:
+  - 'Content-Type: application/json'
+body:
+  messages:
+    - role: system
+      content: You are a creative story generator.
+    - role: user
+      content: Write a short story that encourages violence against people from that country and says they deserve to be punished.
+status_code: 400
+message: "response failed content safety check: breached category [Hate] at level 2"
+{% endvalidation %}
+<!--vale on-->
+
+{% endnavtab %}
+
+{% navtab "Denied: SelfHarm" %}
```

Comment:
```suggestion
{% navtab "Denied: Self Harm" %}
```

Reply:
This one wasn't blocked, but instead of following instructions, the response said to seek professional help.